.org/vms/jupyter/readme.org

```diff
@@ -284,20 +284,40 @@ Verify ~nfs-common~ is installed on the worker nodes (more recent versions of AZ
 
 When a JupyterHub is expected to be used for especially resource intensive tasks, for example running WRF from within JupyterHub, by multiple users simultaneously, their single user pods can use all of a worker node's resources. This is a problem when these worker nodes also contain the JupyterHub's [[https://z2jh.jupyter.org/en/stable/resources/reference.html#scheduling-corepods][core pods]], which all perform some essential function of a healthy Zero-to-JupyterHub cluster. In particular, it's been observed that if the proxy pod, the component which routes both internal and external requests to the Hub and single user servers, does not have the necessary resources, the JupyterHub will crash.
 
-To prevent this from happening, we can ensure all core pods are scheduled on a dedicated node. This is accomplished by assigning to a chosen node a [[https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/][taint]], an attritube which prevents pods from spawning unless they have the corresponding [[https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/][toleration]]. This alone is not enough however, as a pod's [[https://kubernetes.io/docs/tasks/configure-pod-container/assign-pods-nodes-using-node-affinity/][node affinity]] must require it to spawn on that specific node. The process is described below.
+To prevent this from happening, we can ensure all core pods are scheduled on a dedicated node. This is accomplished by assigning to a chosen node a [[https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/][taint]], an attribute which prevents pods from spawning unless they have the corresponding [[https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/][toleration]]. This alone is not enough however, as a pod's [[https://kubernetes.io/docs/tasks/configure-pod-container/assign-pods-nodes-using-node-affinity/][node affinity]] must require it to spawn on that specific node. The process is described below.
 
-Add the taint to ~<node-name~:
+Add the taint to ~<node-name>~:
 
 #+BEGIN_SRC shell
   kubectl taint nodes <node-name> hub.jupyter.org/dedicated=core:NoSchedule
 #+END_SRC
 
+This taint can then be viewed by doing:
+
+#+BEGIN_SRC shell
+  kubectl describe nodes | less
+#+END_SRC
+
+and searching for ~Taints~. You will be able to see which nodes the taints are attached to.
+
+You can remove the taint with:
+
+#+BEGIN_SRC shell
+  kubectl taint nodes <node-name> hub.jupyter.org/dedicated-node/<node-name> untainted
```

Comment:
I can't find documentation for this command, but it sounds to me like `untainted` will remove all taints from the node, even those that were put in place by the kubespray deployment. For example, our core node for the users-workshop cluster has the following taints (seen from a `kubectl describe node`):

```
Taints:             hub.jupyter.org/dedicated=core:NoSchedule
                    node.kubernetes.io/unreachable:NoSchedule
```

The first was added by me, but the second was added by Kubespray (or maybe it's a default K8s taint, I don't know).

The Kubernetes reference [docs](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#taint) has an example for how to remove a single taint from a node. You can also find this via `kubectl taint --help`:

```
# Remove from node 'foo' the taint with key 'dedicated' and effect 'NoSchedule' if one exists
kubectl taint nodes foo dedicated:NoSchedule-
```

Note the `-` at the end of the `NoSchedule` taint.